### PR TITLE
Simplify schedule interface

### PR DESCRIPTION
**Pull request by neeckeloo**

### CODE REVIEW

File: src/Scheduler.php [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+    protected $dateRange;
+
+    /**
+     * {@inheritdoc}
+     */
+    public function schedule(
+        SchedulableEvent $event,
+        TemporalExpressionInterface $temporalExpression
+    ) {
+        if ($this->isScheduled($event)) {
```
Comment (nsitbon) [inline src/Scheduler.php]: one may want to schedule an event more than once using different temporal expression. So either you don't add this check or you add the temporal expression as the second parameter

Comment (neeckeloo) [inline src/Scheduler.php]: Maybe we could allow to add the same event more than once with different temporal expressions. In this scenario, how do you perform the unscheduling of an event? By unscheduling the event and all associate temporal expressions? When we schedule once again an event, we could also use the `Union` temporal expression to compose the existing temporal expression and the new one and keep only one scheduled event in the scheduler.

File: src/DateRange.php [hunk 2]
```diff
@@ -110,14 +93,12 @@ public function getIterator()
      */
     public function getReverseIterator()
     {
-        $date     = clone $this->endDate;
-        $interval = $this->getInterval();
+        $date = $this->endDate;
```
Comment (nsitbon) [inline src/DateRange.php]: for loop ?

File: src/DateRange.php [hunk 3]
```diff
@@ -23,38 +25,41 @@ class DateRange
...
 
     /**
-     * @param DateTime $startDate
-     * @param DateTime $endDate
+     * @param DateTimeInterface $startDate
+     * @param DateTimeInterface $endDate
      * @param DateInterval|null $interval
      */
-    public function __construct(DateTime $startDate, DateTime $endDate, DateInterval $interval = null)
+    public function __construct(DateTimeInterface $startDate, DateTimeInterface $endDate, DateInterval $interval = null)
```
Comment (nsitbon) [inline src/DateRange.php:32]:
``` php
public function __construct(
   DateTimeInterface $startDate, 
   DateTimeInterface $endDate, 
   DateInterval $interval = new DateInterval('P1D')
)
```

File: src/DateRange.php [hunk 4]
```diff
@@ -1,19 +1,21 @@
...
-namespace Riskio\Schedule;
+namespace Riskio\EventScheduler;
 
 use DateInterval;
 use DateTime;
+use DateTimeImmutable;
+use DateTimeInterface;
 use Generator;
 
 class DateRange
```
Comment (nsitbon) [inline src/DateRange.php:10]: what's the difference between this class and the standard DatePeriod class ?

**Pull request merged**